src/main/java/com/aws/greengrass/easysetup/DeviceProvisioningHelper.java

```diff
@@ -350,13 +351,29 @@ private SdkHttpClient getSdkHttpClient() {
      * @param thing         thing info
      * @param awsRegion     aws region
      * @param roleAliasName role alias for using IoT credentials endpoint
+     * @param certPath the path of certificates which users specify
      * @throws IOException                  Exception while reading root CA from file
      * @throws DeviceConfigurationException when the configuration parameters are not valid
      */
     public void updateKernelConfigWithIotConfiguration(Kernel kernel, ThingInfo thing, String awsRegion,
-                                                       String roleAliasName)
+                                                       String roleAliasName, String certPath)
             throws IOException, DeviceConfigurationException {
         Path rootDir = kernel.getNucleusPaths().rootPath();
+
+        if (!Utils.isEmpty(certPath)) {
+            File tmpFile = new File(certPath);
```

Comment:
Do not do any of this with File. Simply create the path using the util method when the path is specified by the user.